src/pytest_workflow/workflow_file_tests.py

```diff
@@ -29,46 +29,105 @@ def __init__(self, name: str, parent: pytest.Collector,
+                 path_existence_pairs: List[Tuple[Path, bool]],
                  cwd: Union[bytes, str]):
         """
         :param name: Name of the test.
         :param parent: Collector that started this test.
-        :param files: A list of paths to be tested.
+        :param path_existence_pairs: A list of paths and whether they should
+        exist.
         :param cwd: The directory relative to which relative paths are tested.
         """
-        self.files = files
+        self.path_existence_pairs = path_existence_pairs
         self.cwd = cwd
         super().__init__(name, parent=parent)
 
     def collect(self):
         """Starts all the file existence tests."""
-        for test_file in self.files:
-            name = "{0}. File exists: {1}".format(self.name, test_file)
-            yield FileExists(name, self, Path(self.cwd) / test_file)
+        for path, should_exist in self.path_existence_pairs:
+            yield FileExists(self.name, self, Path(self.cwd) / path,
+                             should_exist)
 
 
 class FileExists(pytest.Item):
     """A pytest file exists test."""
 
-    def __init__(self, name: str, parent: pytest.Collector, file: Path):
+    def __init__(self, name: str, parent: pytest.Collector, filepath: Path,
+                 should_exist: bool):
         """
         :param name: Test name
         :param parent: Collector that started this test
-        :param file: A path to the file
+        :param filepath: A path to the file
+        :param should_exist: Whether the file should exist
         """
         super().__init__(name, parent)
-        self.file = file
+        self.file = filepath
+        self.should_exist = should_exist
+
+    def runtest(self):
+        assert self.file.exists() == self.should_exist
+
+
+class FilesMd5SumCheckCollector(pytest.Collector, ):
+    def __init__(self, name: str, parent: pytest.Collector,
+                 path_md5_pairs: List[Tuple[Path, str]],
+                 cwd: Union[bytes, str]):
+        super().__init__(name, parent)
+        self.path_md5_pairs = path_md5_pairs
+        self.cwd = cwd
+
+    def collect(self):
+        for path, md5 in self.path_md5_pairs:
+            yield CheckMd5("{0}: check md5sum".format(path), self, path, md5,
+                           self.cwd)
+
+
+class CheckMd5(pytest.Item):
+    def __init__(self, name: str, parent: pytest.Collector, filepath: Path,
+                 md5sum: str, cwd: Union[bytes, str]):
+        super().__init__(name, parent)
+        self.filepath = Path(cwd) / filepath
+        self.md5sum = md5sum
 
     def runtest(self):
-        assert self.file.exists()
+        assert file_md5sum(self.filepath) == self.md5sum
+
+
+def file_md5sum(filepath: Path):
+    """
+    Generates a md5sum for a file. Reads file in blocks to save memory.
+    :param filepath: a pathlib. Path to the file
+    :return: a md5sum as hexadecimal string.
+    """
+    if not filepath.exists():
```

Comment:
This looks somewhat redundant checking. `filepath.open()` already raises a `FileNotFoundError` when the file doesn't exist.